samza-core/src/main/java/org/apache/samza/task/StreamOperatorTask.java

```diff
@@ -39,60 +37,58 @@
 
 /**
  * A {@link StreamTask} implementation that brings all the operator API implementation components together and
- * feeds the input messages into the user-defined transformation chains in {@link StreamApplication}.
+ * feeds the input messages into the user-defined transformation chains in {@link org.apache.samza.operators.StreamGraphImpl}.
  */
 public class StreamOperatorTask implements StreamTask, InitableTask, WindowableTask, ClosableTask {
   private static final Logger LOG = LoggerFactory.getLogger(StreamOperatorTask.class);
 
-  private final StreamApplication streamApplication;
-  private final ApplicationRunner runner;
+  private final SerializedStreamGraph serializedGraph;
   private final Clock clock;
 
   private OperatorImplGraph operatorImplGraph;
   private ContextManager contextManager;
 
   /**
-   * Constructs an adaptor task to run the user-implemented {@link StreamApplication}.
-   * @param streamApplication the user-implemented {@link StreamApplication} that creates the logical DAG
-   * @param runner the {@link ApplicationRunner} to get the mapping between logical and physical streams
+   * Constructs an adaptor task to run the user-implemented {@link org.apache.samza.operators.StreamGraphImpl}.
+   * @param serializedGraph the serialized version of user-implemented {@link org.apache.samza.operators.StreamGraphImpl} that includes the logical DAG
    * @param clock the {@link Clock} to use for time-keeping
    */
-  public StreamOperatorTask(StreamApplication streamApplication, ApplicationRunner runner, Clock clock) {
-    this.streamApplication = streamApplication;
-    this.runner = runner;
+  public StreamOperatorTask(SerializedStreamGraph serializedGraph, Clock clock) {
+    this.serializedGraph = serializedGraph;
     this.clock = clock;
   }
 
-  public StreamOperatorTask(StreamApplication application, ApplicationRunner runner) {
-    this(application, runner, SystemClock.instance());
+  public StreamOperatorTask(SerializedStreamGraph streamGraph) {
```

Comment:
Love this!! This change is really the core of this PR 👍 Do you mind adding a brief description on the JIRA/PR comment? That way, other reviewers have a bit more context.

Reply:
Thanks! I have created a separate design doc to illustrate this change.